src/opentype/tables/simple/cmap.js

```diff
@@ -39,6 +39,24 @@ class cmap extends SimpleTable {
     return subtable.getSupportedCharCodes();
   }
 
+  reverse(glyphid) {
+    for(let i=0; i<this.numTables; i++) {
+      let code = this.getSubTable(i).reverse(glyphid);
+      if (code) return code;
+    }
+  }
+
+  getGlyphId(char) {
```

Comment:
This seems to always return either `0` or `undefined`, tripping up [this call](https://github.com/Pomax/Font.js/pull/89/files#diff-8c6819ac5d3de8c5283fc7619e571e4cR85). The `!== 0` returns true for `undefined`, false for `0`.

Reply:
if it always returns 0 or undefined, then either there's a bug, or the font doesn't actually support anythying... case 1 is more likely, so we'll need to check why `last` is not getting assigned. From the code, the most likely culprit is `t.getGlyphId` not existing for subtables.

Reply:
I traced this as far as I could, and ended up with a possible smoking gun [here](https://github.com/Pomax/Font.js/pull/89/files#diff-8c6819ac5d3de8c5283fc7619e571e4cR59). Most of these values turn out to be zero when I log `glyphIDs`:

```
[
  0, 0,   0,   0, 186, 0, 0,   0, 0, 0, 0,
  0, 0, 190,   0,   0, 0, 0,   0, 0, 0, 0,
  0, 0,   0,   0,   0, 0, 0,   0, 0, 0, 0,
  0, 0,   0, 152,   0, 0, 0, 172, 0, 0, 0,
  0, 0,   0,   0,   0, 0, 0,   0
]
```

I'd expect them to be unique IDs?

Reply:
yeah, that does not look like a correct segment at all O_o! which font is this with?

Reply:
I'm seeing this with OpenSans-Regular.ttf. If I put back Recursive in the test.js, I get more sensible looking IDs, by the way!

Reply:
perfect. I might have time before Friday to look into this, but if not, Friday it is.

Reply:
Awesome, let me know if I can test or help!

Reply:
Looks like this is our good friend "signed vs. unsigned"

![image](https://user-images.githubusercontent.com/177243/95637333-4f56a900-0a46-11eb-87cc-92eac9fee6e3.png)

![image](https://user-images.githubusercontent.com/177243/95637343-54b3f380-0a46-11eb-83ff-1bf9d0da3d95.png)

Reply:
Although that does not seem to fully fix things:

![image](https://user-images.githubusercontent.com/177243/95637400-92b11780-0a46-11eb-9b00-8a952ad8cb31.png)

Reply:
The calculation in https://github.com/Pomax/Font.js/pull/89/files#diff-8c6819ac5d3de8c5283fc7619e571e4cR59-R62 is probably at fault here - the spec is.. really weird in its description of how to write this algorithm, relying on people understanding what C/C++'s `&` and `*` mean.

Reply:
Updated the reverse lookup to what looks like the correct thing now - I tested this using `node test\gsub\test.segment.js` and get this for the first five segments now:
```
{
  startCode: 32,
  endCode: 72,
  idDelta: -29,
  idRangeOffset: 0,
  glyphIDs: [
     3,  4,  5,  6,  7,  8,  9, 10, 11, 12,
    13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
    23, 24, 25, 26, 27, 28, 29, 30, 31, 32,
    33, 34, 35, 36, 37, 38, 39, 40, 41, 42,
    43
  ]
}
{
  startCode: 73,
  endCode: 73,
  idDelta: 0,
  idRangeOffset: 174,
  glyphIDs: [ 918 ]
}
{
  startCode: 74,
  endCode: 126,
  idDelta: -29,
  idRangeOffset: 0,
  glyphIDs: [
    45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56,
    57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68,
    69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80,
    81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92,
    93, 94, 95, 96, 97
  ]
}
{
  startCode: 160,
  endCode: 203,
  idDelta: -62,
  idRangeOffset: 0,
  glyphIDs: [
     98,  99, 100, 101, 102, 103, 104, 105,
    106, 107, 108, 109, 110, 111, 112, 113,
    114, 115, 116, 117, 118, 119, 120, 121,
    122, 123, 124, 125, 126, 127, 128, 129,
    130, 131, 132, 133, 134, 135, 136, 137,
    138, 139, 140, 141
  ]
}
{
  startCode: 204,
  endCode: 207,
  idDelta: 0,
  idRangeOffset: 170,
  glyphIDs: [ 919, 920, 921, 922 ]
}
```

Reply:
I'm curious to know why glyphid 44 seems missing...